### PR TITLE
Fix that options --upload-only & --remove-source-files are ignored on an upload session restore

### PR DESCRIPTION
**Pull request by abraunegg**
* Fix that options --upload-only & --remove-source-files are ignored on an upload session restore

### CODE REVIEW

File: src/sync.d [hunk 1]
```diff
@@ -445,9 +445,36 @@ final class SyncEngine
 			// Check if there is an interrupted upload session
 			if (session.restore()) {
 				log.log("Continuing the upload session ...");
+				string sessionFilePath = session.getSessionFilePath();
 				auto item = session.upload();
-				saveItem(item);
-			}		
+				
+				if (!uploadFailed) {
```
Comment (SylvainLaugie) [inline src/sync.d:451]: Is `uploadFailed` ever going to be true here?

File: src/upload.d [hunk 2]
```diff
@@ -268,6 +268,15 @@ struct UploadSession
 		}
 	}
 	
+	string getSessionFilePath() {
```
Comment (SylvainLaugie) [inline src/upload.d:271]: Should you name this method something like "getSessionLocalPath"?  
`sessionFilePath` is an existing member attribute, took me some time to understand that this new `getSessionFilePath()` doesn't return `sessionFilePath`

File: src/sync.d [hunk 3]
```diff
@@ -445,9 +445,36 @@ final class SyncEngine
 			// Check if there is an interrupted upload session
 			if (session.restore()) {
 				log.log("Continuing the upload session ...");
+				string sessionFilePath = session.getSessionFilePath();
```
Comment (SylvainLaugie) [inline src/sync.d:448]: About this variable name: see comment on upload.d https://github.com/abraunegg/onedrive/pull/1399/files#r611176161

File: src/sync.d (hunk 1 above)
Comment (abraunegg) [reply src/sync.d:451]: If an upload fails, using the functions that session.upload() utilities , there is the potential for this to be flagged as true.

File: src/sync.d (hunk 3 above)
Comment (abraunegg) [reply src/sync.d:448]: good point .. yes .. function name should not be named the same

File: src/sync.d (hunk 1 above)
Comment (abraunegg) [reply src/sync.d]: Resolved by https://github.com/abraunegg/onedrive/pull/1399/commits/9cdf81b074715c9f07a5be92d99d7bf5c22cc324

File: src/sync.d (hunk 3 above)
Comment (abraunegg) [reply src/sync.d]: Resolved by https://github.com/abraunegg/onedrive/pull/1399/commits/9cdf81b074715c9f07a5be92d99d7bf5c22cc324

File: src/upload.d (hunk 2 above)
Comment (abraunegg) [reply src/upload.d]: Resolved by https://github.com/abraunegg/onedrive/pull/1399/commits/9cdf81b074715c9f07a5be92d99d7bf5c22cc324

**Pull request merged**